yarn/src/main/scala/org/apache/spark/deploy/yarn/security/AMDelegationTokenRenewer.scala

```diff
@@ -69,6 +71,9 @@ private[yarn] class AMDelegationTokenRenewer(
   private val freshHadoopConf =
     hadoopUtil.getConfBypassingFSCache(hadoopConf, new Path(credentialsFile).toUri.getScheme)
 
+  @volatile private var timeOfNextRenewal =
+    sparkConf.get(CREDENTIALS_RENEWAL_TIME).getOrElse(Long.MaxValue)
```

Comment:
From my understanding you should never get to this code when  `CREDENTIALS_RENEWAL_TIME` is not set, so you should use `.get` to force and exception in case the config is not set.

If indeed that situation can occur and you really need a default value, it should be declared in the constant itself and not here where it's used.